src/Repository/ApActivityRepository.php

```diff
@@ -27,12 +29,55 @@ public function __construct(ManagerRegistry $registry, private SettingsManager $
         parent::__construct($registry, ApActivity::class);
     }
 
+    #[ArrayShape([
+        'id' => 'int',
+        'type' => 'string',
+    ])]
     public function findByObjectId(string $apId): ?array
+    {
+        $local = $this->findLocalByApId($apId);
+        if ($local) {
+            return $local;
+        }
+
+        $conn = $this->_em->getConnection();
+        $tables = [
+            ['table' => 'entry', 'class' => Entry::class],
+            ['table' => 'entry_comment', 'class' => EntryComment::class],
+            ['table' => 'post', 'class' => Post::class],
+            ['table' => 'post_comment', 'class' => PostComment::class],
+        ];
+        foreach ($tables as $table) {
```

Comment:
Maybe a single query instead of looping over multiple tables could be faster/more efficient for postgresql.

Reply:
It previously was a union. I don't know whether it would be faster, but I don't think that this is slower, since it could happen that we only execute one or two of the queries if it finds the object.

Reply:
It may have a bit to do with how often it has to establish a connection, as that can be one of the more expensive things compared to a query. Not sure if we have connection pooling to postgres already; if not we might want to put pgbouncer on the list just like amqproxy

Reply:
we have a connection pool (see `config/packages/doctrine.yaml`), but still:

```
    doctrine:
        orm:
            auto_generate_proxy_classes: false
            proxy_dir: '%kernel.build_dir%/doctrine/orm/Proxies'
            query_cache_driver:
                type: pool
                pool: doctrine.system_cache_pool
            result_cache_driver:
                type: pool
                pool: doctrine.result_cache_pool

    framework:
        cache:
            pools:
                doctrine.result_cache_pool:
                    adapter: cache.app
                doctrine.system_cache_pool:
                    adapter: cache.system
```

Reply:
But maybe we are missing `default_doctrine_dbal_provider: 'doctrine.dbal.default_connection'`? https://symfony.com/doc/current/cache.html

Reply:
We might want to add a PDO cache adapter: `default_pdo_provider` (shortcut)

Reply:
but the connection is created outside of the foreach...

Reply:
I just answer the question of @e-five256 . But I will close this review comment. If you want to do multiple queries in a for loop. OK.